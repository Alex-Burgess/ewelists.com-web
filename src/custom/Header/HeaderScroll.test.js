import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import HeaderScroll from './HeaderScroll';

test('Renders HeaderFixed component, when user is not authenticated.', () => {
  const tree = renderer
    .create(
      <Router>
        <HeaderScroll isAuthenticated={false} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders HeaderFixed component, when user is authenticated.', () => {
  const tree = renderer
    .create(
      <Router>
        <HeaderScroll isAuthenticated={true} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
