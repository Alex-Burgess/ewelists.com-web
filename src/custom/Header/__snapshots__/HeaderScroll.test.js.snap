// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HeaderFixed component, when user is authenticated. 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-21 makeStyles-transparent-33 makeStyles-fixed-23 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-container-24 MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-25"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="logo"
              className="makeStyles-logo-20"
              src="logo-white.png"
            />
          </a>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <div
        className="makeStyles-hidden-37 PrivateHiddenCss-smDown-119"
      >
        <div
          className="makeStyles-collapse-38"
        >
          <ul
            className="MuiList-root makeStyles-list-171 makeStyles-mlAuto-187 MuiList-padding"
          >
            <li
              className="MuiListItem-root makeStyles-listItem-172 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="makeStyles-navLink-174"
                href="/"
                onClick={[Function]}
              >
                My Lists
              </a>
            </li>
            <li
              className="MuiListItem-root makeStyles-listItem-172 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="makeStyles-navLink-174"
                href="/list-ideas"
                onClick={[Function]}
              >
                Gift List Ideas
              </a>
            </li>
            <li
              className="MuiListItem-root makeStyles-listItem-172 MuiListItem-gutters"
              disabled={false}
            >
              <div
                className="makeStyles-manager-190"
              >
                <div
                  className="makeStyles-target-192"
                >
                  <button
                    aria-haspopup="true"
                    aria-label="Notifications"
                    aria-owns={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-213 makeStyles-transparent-236 makeStyles-navLink-174"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root makeStyles-buttonIcon-206"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h20v20H0V0z"
                          fill="none"
                          transform="scale(1.2, 1.2)"
                        />
                        <path
                          d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                          transform="scale(1.2, 1.2)"
                        />
                      </svg>
                      <b
                        className="makeStyles-caret-207"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Renders HeaderFixed component, when user is not authenticated. 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-21 makeStyles-transparent-33 makeStyles-fixed-23 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-container-24 MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-25"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="logo"
              className="makeStyles-logo-20"
              src="logo-white.png"
            />
          </a>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <div
        className="makeStyles-hidden-37 PrivateHiddenCss-smDown-119"
      >
        <div
          className="makeStyles-collapse-38"
        >
          <ul
            className="MuiList-root makeStyles-list-129 makeStyles-mlAuto-145 MuiList-padding"
          >
            <li
              className="MuiListItem-root makeStyles-listItem-130 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="makeStyles-navLink-132"
                href="/list-ideas"
                onClick={[Function]}
              >
                Gift List Ideas
              </a>
            </li>
            <li
              className="MuiListItem-root makeStyles-listItem-130 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="makeStyles-navLink-132"
                href="/login"
                onClick={[Function]}
              >
                Log In
              </a>
            </li>
            <li
              className="MuiListItem-root makeStyles-listItem-130 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="makeStyles-navLink-132"
                href="/signup"
                onClick={[Function]}
              >
                Sign Up
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
</div>
`;
